File: SpringBootCors/src/main/java/com/example/java/web/MessageController.java
package com.example.java.web;

import com.example.java.domain.Message;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("message")
public class MessageController {

	//@CrossOrigin
	@GetMapping("{value}")
	public String get(@PathVariable String value) {
		return value;
	}

	@PostMapping
	public Message post(@RequestBody Message message) {
		return message;
	}
}

File: SpringBootCors/src/main/kotlin/com/example/kotlin/web/MessageController.kt
package com.example.kotlin.web

import com.example.kotlin.domain.Message
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("message")
class MessageController {

    //@CrossOrigin
    @GetMapping("{value}")
    fun get(@PathVariable value: String): String {
        return value
    }

    @PostMapping
    fun post(@RequestBody message: Message): Message {
        return message
    }
}
